mask-payload/mask.js

```diff
@@ -0,0 +1,124 @@
+    return crypto.createHash("sha256").update(data).digest("hex");
+  } catch (err) {
+    if (err.message.includes("ENOENT")) {
+      return "";
+    }
+    throw err;
+  }
+}
+
+function envWrite({ payload, fd, varname, path, backup = "" }) {
+  let toWrite = payload;
+  path.split(".").forEach((arg) => {
+    toWrite = toWrite[arg] ?? {};
+  });
+  if (!["string", "boolean", "number"].some((type) => typeof toWrite === type)) {
+    toWrite = backup;
+  }
+  fs.writeSync(fd, Buffer.from(`${varname}=${toWrite ?? ""}\n`));
+}
+
+function run() {
+  try {
+    const inputs = JSON.parse(process.env.MASK_INPUTS ?? "{}");
+    const githubEventPR = JSON.parse(process.env.MASK_GITHUB_EVENT_PR ?? "{}");
+
+    const filepath = process.env.GITHUB_EVENT_PATH;
+    let payload = {};
+    if (inputs["check-mode"] === "payload" && filepath) {
+      const event = JSON.parse(fs.readFileSync(filepath).toString());
+      payload = JSON.parse(event?.inputs?.payload) ?? {};
+    }
+
+    const githubEnv = fs.openSync(process.env.GITHUB_ENV, "a");
+    const githubToken = payload?.githubToken ?? inputs["githubToken"] ?? "";
+    const trunkToken = payload?.trunkToken ?? inputs["trunkToken"] ?? "";
+
+    process.stdout.write(`::add-mask::${githubToken}\n`);
+    process.stdout.write(`::add-mask::${trunkToken}\n`);
+    fs.writeSync(githubEnv, Buffer.from(`INPUT_GITHUB_TOKEN=${githubToken}\n`));
+    fs.writeSync(githubEnv, Buffer.from(`INPUT_TRUNK_TOKEN=${trunkToken}\n`));
+    fs.writeSync(githubEnv, Buffer.from(`TRUNK_TOKEN=${trunkToken}\n`));
+
+    const envVarConfigs = [
+      {
+        varname: "GITHUB_EVENT_PULL_REQUEST_BASE_REPO_OWNER",
+        path: "pullRequest.base.repo.owner.login",
+        backup: "",
+      },
+      {
+        varname: "GITHUB_EVENT_PULL_REQUEST_BASE_REPO_NAME",
+        path: "pullRequest.base.repo.name",
+        backup: "",
+      },
+      {
+        varname: "GITHUB_EVENT_PULL_REQUEST_BASE_SHA",
+        path: "pullRequest.base.sha",
+        backup: githubEventPR?.base?.sha ?? "",
+      },
+      {
+        varname: "GITHUB_EVENT_PULL_REQUEST_HEAD_REPO_FORK",
+        path: "pullRequest.head.repo.fork",
+        backup: githubEventPR?.head?.repo?.fork ?? "",
+      },
+      {
+        varname: "GITHUB_EVENT_PULL_REQUEST_HEAD_SHA",
+        path: "pullRequest.head.sha",
+        backup: githubEventPR?.head?.sha ?? "",
+      },
+      {
+        varname: "GITHUB_EVENT_PULL_REQUEST_NUMBER",
+        path: "pullRequest.number",
+        backup: githubEventPR?.number ?? "",
+      },
+      { varname: "GITHUB_REF_NAME", path: "targetRefName", backup: process.env.GITHUB_REF_NAME },
+      { varname: "INPUT_ARGUMENTS", path: "arguments", backup: inputs["arguments"] },
+      { varname: "INPUT_CACHE", path: "cache", backup: inputs["cache"] },
+      {
+        varname: "INPUT_CACHE_KEY",
+        path: "cacheKey",
+        backup: `trunk-${inputs["cache-key"]}-${process.env.RUNNER_OS}-${hashFile(
```

Comment:
```suggestion
        backup: `trunk-${inputs.`cache-key`}-${process.env.RUNNER_OS}-${hashFile(
```

Reply:
This syntax doesn't seem to work for me?